apps/dotcom/sync-worker/src/TLPostgresReplicator.ts

```diff
@@ -187,6 +190,27 @@ export class TLPostgresReplicator extends DurableObject<Environment> {
 				this.db
 			)
 		}
+		this.maybePruneUsers()
+	}
+
+	private async maybePruneUsers() {
+		const now = Date.now()
+		if (now - this.lastUserPruneTime < PRUNE_TIME) return
+		this.lastUserPruneTime = now
+		const activeUserIds = this.ctx.storage.sql
+			.exec('SELECT id FROM active_user')
+			.toArray()
+			.map((u) => u.id) as string[]
+		for (const id of activeUserIds) {
+			const lastUserUpdate = this.usersWithUpdates.get(id)
+			// User has had updates in the last prune period, so skip them
+			if (lastUserUpdate && now - lastUserUpdate < PRUNE_TIME) continue
+			// Their time is up
+			this.usersWithUpdates.delete(id)
```

Comment:
should this line be moved into `unregisterUser`?

Reply:
Their time is up in any case, so I think we should just evict them here. The only thing that will change is that it will short circuit the above if condition.

Reply:
but if they're still active but we don't send them another update, they'll always be short-circuited right?